app/src/androidTest/java/Cryptohopper/Android/SDK/user/UserInstrumentedTest.kt

```diff
@@ -0,0 +1,312 @@
+package Cryptohopper.Android.SDK.exchange
+
+import Cryptohopper.Android.SDK.helper.Const
+import Cryptohopper.Android.SDK.helper.Const.API_KEY
+import Cryptohopper.Android.SDK.helper.Const.API_PASSWORD
+import Cryptohopper.Android.SDK.helper.Const.API_USER
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.mervick.aes_everywhere.Aes256
+import cryptohopper.android.sdk.CryptohopperAuth
+import cryptohopper.android.sdk.SharedModels.ConfigModels.HopperAPIEnvironment
+import CryptohopperUser
+import kotlinx.coroutines.*
+import org.junit.Assert
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.Before
+import kotlin.random.Random
+
+@DelicateCoroutinesApi
+@RunWith(AndroidJUnit4::class)
+class UserInstrumentedTest {
+
+    @Before
+    fun setup() {
+        val appContext = InstrumentationRegistry.getInstrumentation().targetContext
+        CryptoHopperConfig.configure(
+            appContext, API_KEY,
+            HopperAPIEnvironment.Production
+        )
+        callAuthenticationWithAccurateDetails()
+    }
+
+    private fun callAuthenticationWithAccurateDetails() {
+        val userAgent = Aes256.encrypt(API_USER, Const.API_AGENT)
+        CryptohopperAuth.login(
+            username = API_USER,
+            password = API_PASSWORD,
+            verificationCode = null,
+            userAgent = userAgent
+        ) { result, error ->
+            Assert.assertNull(error)
+            Assert.assertNotNull(result)
+        }
+    }
+
+    //******************* Positive cases ***************************
+
+    @Test
+    fun when_the_given_getUserProfile_Endpoint_is_called_with_correct_token_then_it_must_return_profile_details() {
+        CryptohopperUser.getUserProfile { result, error ->
+            Assert.assertNull(error)
+            Assert.assertNotNull(result)
+
+            Assert.assertNotNull(result?.name!!)
+            Assert.assertNotNull(result.username)
+            Assert.assertNotNull(result.email)
+            Assert.assertNotNull(result.website)
+            Assert.assertNotNull(result.country)
```

Comment:
Same , if something is empty in the database , API is not sending it as "" but instead not sending it which results in null